aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -196,22 +198,19 @@ Tensor empty_like(
     return result;
   }
 
-  auto memory_format =
-      optional_memory_format.value_or(MemoryFormat::Contiguous);
-  auto use_memory_format = memory_format;
-  if (memory_format == MemoryFormat::Preserve) {
-    if (self.is_contiguous(MemoryFormat::ChannelsLast)) {
-      use_memory_format = MemoryFormat::ChannelsLast;
-    } else if (self.is_contiguous(MemoryFormat::Contiguous)) {
-      use_memory_format = MemoryFormat::Contiguous;
-    } else {
-      TORCH_CHECK(
-          false,
-          "undefined behavior of the preserve format, source tensor neither channels last nor contiguous")
+  if (self.is_quantized()) {
+
+    auto memory_format =
+        optional_memory_format.value_or(MemoryFormat::Contiguous);
```

Comment:
To be changed later right :>

Reply:
Yes, right now we need to introduce this option, after we check all current code for expected behaviour and change default.

Reply:
If it's to be changed later, can we add a comment linking some issue or with some information ?

Reply:
What Richard said